src/devices/interlogix.c

```diff
@@ -0,0 +1,404 @@
+
+                //set message starting postion (just past preamble and sync bit)
+                bit_offset = sizeof(preamble)*8 + bit_offset;
+
+                uint8_t message[6];
+                unsigned msgLength = bitbuffer->bits_per_row[row]-bit_offset+2; //TODO: what happens if a different message size arrives?
+                if (debug_output >= 2)
+                        fprintf(stderr, "Message length: %d \n", msgLength);
+
+                bitbuffer_extract_bytes(bitbuffer, row, bit_offset, message, msgLength);
+
+
+                //TODO: Check message integrity (CRC/Checksum/parity) using crc8
+                //      message does not include sync bit that needs to be included in CRC calculation
+                //      missing 2 trailing bits in protocol that include CRC and stop bit?
+                if (crc8(message, 5, INTERLOGIX_CRC_POLY, INTERLOGIX_CRC_INIT) != message[5]) {
+                        if (debug_output >= 1)
+                                fprintf(stderr, "CRC Check for message payload failed! \n");
+                        //return 0;
+                }
+
+                if (debug_output >= 1) {
+
+                        //grab 6, 4 bit nibbles from bit buffer. need to reverse them to get serial number
+                        fprintf(stderr, "\nDevice Serial Number: ");
+                        fprintf(stderr, "%02x", reverse8(message[2]));
+                        fprintf(stderr, "%02x", reverse8(message[1]));
+                        fprintf(stderr, "%02x", reverse8(message[0]));
+                        fprintf(stderr, "\n");
+
+                        fprintf(stderr, "Device Message: ");
+                        printBits(sizeof(message[0]), &message[0]);
+                        printBits(sizeof(message[1]), &message[1]);
+                        printBits(sizeof(message[2]), &message[2]);
+                        printBits(sizeof(message[3]), &message[3]);
+                        printBits(sizeof(message[4]), &message[4]);
+                        printBits(sizeof(message[5]), &message[5]);
+                        fprintf(stderr, "\n");
+
+                }
+
+                sprintf(device_type, "%01x", (reverse8(message[2])>>4));
+
+                //TODO: look at conditionally parsing message for specific device types - raw message
+                //      will be passed so parsing for specific device can be done at higher layer
+                switch((reverse8(message[2])>>4)) {
+                        case 0xa: device_type_name = "contact sensor"; break;
+                        case 0xf: device_type_name = "keyfob"; break;
+                        case 0x4: device_type_name = "motion sensor"; break;
+                        case 0x6: device_type_name = "heat sensor"; break;
+
+                        default: device_type_name = "unknown"; break;
+                }
+
+
+
+
+                low_battery = (message[3] & 0x10) ? "on" : "off";
```

Comment:
Look at the other decoders for usual keys. `"battery",       "Battery",     DATA_STRING, battery_low ? "LOW" : "OK",`